Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -2182,10 +2198,19 @@ void completeWithBlockedError()
     config.get().sourceWebRegistrableDomain = pcm.sourceSite().registrableDomain.string();
     config.get().version = @"4.0";
     config.get().attributionContext = AttributionTypeDefault;
+#if HAVE(AD_ATTRIBUTION_KIT_PRIVATE_BROWSING)
+    if (m_sessionID.isEphemeral()) {
+        m_donatedEphemeralImpressionSessionID = WTF::UUID::createVersion4();
+
+        // FIXME: Remove this respondsToSelector check in 2026 or so.
+        if ([config respondsToSelector:@selector(privateBrowsingSessionId)])
+            config.get().privateBrowsingSessionId = *m_donatedEphemeralImpressionSessionID;
+    }
+#endif // HAVE(AD_ATTRIBUTION_KIT_PRIVATE_BROWSING)
 #if HAVE(ASD_INSTALL_WEB_ATTRIBUTION_SERVICE)
     [[ASDInstallWebAttributionService sharedInstance] addInstallWebAttributionParamsWithConfig:config.get() completionHandler:^(NSError *) { }];
-#endif
-#endif
+#endif // HAVE(ASD_INSTALL_WEB_ATTRIBUTION_SERVICE)
+#endif // HAVE(SKADNETWORK_v4)
```

Comment:
Do we have other references to SKAdNetwork here, and this is just a clarification of something old? Or should we be using the AdAttributionKit name instead? (v4 sounds old to me so may be fine)

Reply:
This is old.  We should come by later and clean it up.